_getline.c

```diff
@@ -0,0 +1,52 @@
+#include "shell.h"
+
+/**
+ * _getline - reads an entire line from fd, storing the address of
+ *the buffer containing the text into *lineptr.
+ * @fd: the stream.
+ * @lineptr: the buffer containing the line from fd.
+ * @n: the size of the buffer.
+ *
+ * Return: number of characters read, or -1 on failure.
+ */
+
+ssize_t _getline(char **lineptr, size_t *n, int fd)
+{
+	char *new_ptr;
+	static size_t count;
+
+	if (lineptr == NULL || n == NULL)
+		return (-1); /* error indicating invalid arguments */
+
+	if (*lineptr == NULL || *n == 0)
+	{
+		*n = INIT_SIZE;
+		*lineptr = malloc(*n);
+
+		if (*lineptr == NULL)
+			return (-1); /* error indicating the failure of malloc */
+	}
+
+	while ((count += read(fd, *lineptr + count, INIT_SIZE)) != 0)
```

Comment:
i just wanted to mimic the behavior of the standard getline(), i can't resize memory in stack on the runtime.
but if we define a buffer of size 1024, it will be enough to carry a line, is that what you meant ?

Reply:
no, we need to read the data fast and use fewer calls to read, so we will read 1024 chars, to be fast for read we will store data in the local string buffer, and the dynamic var will store only the remaining chars.
we don't need to resize stack var it is impossible, the stack var is used to read data using the read `function`.